File: Frontend/src/events/emulator-connection-error-codes.ts
export const EmulatorConnectionErrorCodes = {
    ConfigMissing: "config_missing",
    ProcessNotFound: "process_not_found",
    MappingNotFound: "mapping_not_found",
    ConnectionFailed: "connection_failed",
    MemoryReadFailed: "memory_read_failed",
    StateComposeFailed: "state_compose_failed",
} as const;

export type EmulatorConnectionErrorCode =
    (typeof EmulatorConnectionErrorCodes)[keyof typeof EmulatorConnectionErrorCodes];

const knownErrorCodes = new Set<string>(Object.values(EmulatorConnectionErrorCodes));

export function isKnownEmulatorConnectionErrorCode(value: string): value is EmulatorConnectionErrorCode {
    return knownErrorCodes.has(value);
}

File: Backend/Infrastructure/Duckstation/EmulatorConnectionErrorCodes.cs
namespace Backend.Infrastructure.Duckstation;

public static class EmulatorConnectionErrorCodes
{
    public const string ConfigMissing = "config_missing";
    public const string ProcessNotFound = "process_not_found";
    public const string MappingNotFound = "mapping_not_found";
    public const string ConnectionFailed = "connection_failed";
    public const string MemoryReadFailed = "memory_read_failed";
    public const string StateComposeFailed = "state_compose_failed";
}
